plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/eap/OpenShiftEapDeployPathController.java

```diff
@@ -33,16 +40,40 @@ protected ModuleDeploymentPrefsUtil createModuleDeploymentPrefsUtil() {
 		// We must override the presumed name given to the module upon publish if it is our magic project
 		@Override
 		protected String getOutputNameFromSettings(IServerAttributes server, IModule module) {
+		    try {
+		        Method clearCacheMethod = J2EEDeployableFactory.class.getDeclaredMethod("clearCache", IProject.class);
+		        clearCacheMethod.setAccessible(true);
+		        clearCacheMethod.invoke(JEEDeployableFactory.jeeInstance(), getEclipseProject(server));
+            } catch (IllegalAccessException
+                    | IllegalArgumentException
+                    | InvocationTargetException
+                    | NoSuchMethodException
+                    | SecurityException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
 			String ret = super.getOutputNameFromSettings(server, module);
 			if (ret == null && module.equals(findProjectModule(server))) {
 				String suffix = ServerModelUtilities.getDefaultSuffixForModule(module);
-				ret = "ROOT" + suffix;
+				String prefix = "";
+				if (module instanceof IModule2) {
+				    prefix = ((IModule2)module).getProperty(IModule2.PROP_DEPLOY_NAME);
+		        }
+		        // Otherwise use the module's name
+		        if( prefix == null ) {
+		            prefix = module.getName();
+		        }
+				ret = prefix + suffix;
 			}
 			return ret;
 		}
 
 		protected IModule findProjectModule(IServerAttributes server) {
-			return OpenShiftServerUtils.findProjectModule(OpenShiftServerUtils.getDeployProject(server));
+			return OpenShiftServerUtils.findProjectModule(getEclipseProject(server));
+		}
+		
+		private IProject getEclipseProject(IServerAttributes server) {
```

Comment:
any particular reason for extracting the call to the util method to it's own *private* method?